DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs

```diff
@@ -0,0 +1,182 @@
+﻿using DocumentService.Azure;
+using DocumentService.Contexts;
+using DocumentService.Controllers;
+using DocumentService.Models;
+using DocumentService.Repositories;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+using System.Net;
+
+namespace DocumentService.Unit.Tests.Controllers
+{
+    public class DocumentsControllerTests 
+    {
+        private readonly Mock<IDocumentContext> documentContext;
+        private readonly Mock<IDocumentRepository> documentRepository;
+        private readonly Mock<IAzureBlobService> azureBlobService;
+        private readonly Mock<IConfiguration> configuration;
+
+        public DocumentsControllerTests()
+        {
+            this.documentContext = new Mock<IDocumentContext>();
+            this.documentRepository = new Mock<IDocumentRepository>();
+            this.azureBlobService = new Mock<IAzureBlobService>();
+            this.configuration = new Mock<IConfiguration>();
+        }
+
+        [Fact]
+        public void GetDocumentById_True_WhenIdExists()
+        {
+            // Arrange
+            var guid = Guid.NewGuid();
+            var documentController = new DocumentsController(this.documentRepository.Object, this.azureBlobService.Object, this.configuration.Object);
+
+            DocumentInfo documentInfo = new DocumentInfo
+            {
+                DocumentId = guid,
+                DateCreated = DateTime.UtcNow,
+                Description = "Generic Description",
+                FileName = "Test Doc",
+                DocumentTypes = new DocumentTypes { DocumentType = "Test", DocumentTypesId = 0 },
+                IsDeleted = false
+            };
+            documentRepository.Setup(x => x.GetDocumentAsync(guid)).Returns(Task.FromResult(documentInfo));
+
+            // Act
+            var response = documentController.GetDocumentById(guid);
+            var res = response as OkObjectResult;
+            dynamic result = res.Value;
+            var docInfo = (DocumentInfo)result.GetType().GetProperty("document").GetValue(result, null);
```

Comment:
```suggestion
            var docInfo = (DocumentDTO)result.GetType().GetProperty("document").GetValue(result, null);
```
I suggest we cast the result to a DocumentDTO